core/beneficiary/saver.go

```diff
@@ -61,23 +63,77 @@ type l1Saver struct {
 	set     settler
 	ad      addressProvider
 	chainID int64
+	beneficiaryChangeKeeper
 }
 
-func newL1Saver(chainID int64, ad addressProvider, bc multiChainBC, set settler) *l1Saver {
+func newL1Saver(chainID int64, ad addressProvider, set settler, st storage) *l1Saver {
 	return &l1Saver{
-		chainID: chainID,
-		set:     set,
-		ad:      ad,
+		chainID:                 chainID,
+		set:                     set,
+		ad:                      ad,
+		beneficiaryChangeKeeper: beneficiaryChangeKeeper{st: st, chainID: chainID},
 	}
 }
 
+// SettleWithBeneficiaryState represents the state of settle with beneficiary transaction
+type SettleWithBeneficiaryState string
+
+const bucketBeneficiaryChangeStatus = "beneficiary-change-status"
+const (
+	// PENDING transaction is pending
+	PENDING SettleWithBeneficiaryState = "pending"
```

Comment:
Standard library uses camel-case for constants or any other variable except maybe for some vars in `os` package, so I would suggest to keep it like that.